src/components/SingletonEditor.tsx

```diff
@@ -93,7 +93,10 @@ export const useSingletonEditor = (
             type: "unlayer",
           };
 
-      onSave(content as Content);
+      if (currentUpdateCounter >= savedCounter.current) {
+        onSave(content as Content);
+        savedCounter.current = currentUpdateCounter;
```

Comment:
```suggestion
        savedCounter.current = currentUpdateCounter;
        onSave(content as Content);
```

Que opinás de intercambiar el orden de estas líneas?

Ahora mismo no genera diferencia en el comportamiento ya que `onSave` es un _fire and forget_, pero si llegamos a hacer que `onSave` sea _awaitable_ podría darse un _race condition_ y que a `savedCounter.current` se le asigne un valor anterior.

Ejemplo

```js
      if (currentUpdateCounter >= savedCounter.current) {
        await onSave(content as Content);
        savedCounter.current = currentUpdateCounter;
```

<img width="1259" alt="image" src="https://user-images.githubusercontent.com/1157864/171419315-d1bfb045-28af-4b9a-8ab2-6d30dbf61ce1.png">

Es cierto que de darse este escenario podríamos tener problemas más graves ya que no sabemos realmente que `onSave` fue el último en aplicarse en el backend, pero de cualquier manera me parece correcto actualizar antes.

Reply:
Estoy de acuerdo 👍🏻